client/auth.go

```diff
@@ -211,8 +211,15 @@ func (c *Conn) writeAuthHandshake() error {
 	capability := mysql.CLIENT_PROTOCOL_41 | mysql.CLIENT_SECURE_CONNECTION |
 		mysql.CLIENT_LONG_PASSWORD | mysql.CLIENT_TRANSACTIONS | mysql.CLIENT_PLUGIN_AUTH
 	// Adjust client capability flags based on server support
-	capability |= c.capability & mysql.CLIENT_LONG_FLAG
-	capability |= c.capability & mysql.CLIENT_QUERY_ATTRIBUTES
```

Comment:
I think we don't need to specially process CLIENT_QUERY_ATTRIBUTES. There are 3 states of capability:

1. developer calls `SetCapability`, which means the `ccaps` bit is 1 and the `dcaps` bit is 0
2. developer calls `UnsetCapability`, which means `ccaps` bit is 0 and the `dcaps` bit is 1
3. No calls, which means both bits are 0

So here we turn on the bits by `ccaps` at around line 219, and have another round to turn off bits by `dcaps`. I suggest

```
capability |= c.ccaps&mysql.CLIENT_FOUND_ROWS | ...
capability &= c.dcaps... // here we add CLIENT_QUERY_ATTRIBUTES to turn off